library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileOutputOperator.java

```diff
@@ -636,7 +638,7 @@ protected FSDataOutputStream openStream(Path filepath, boolean append) throws IO
   {
     FSDataOutputStream fsOutput;
     if (append) {
-      fsOutput = fs.append(filepath);
+      return openStreamForNonAppendFS(filepath);
```

Comment:
How about 
fsOutput = openStreamForNonAppendFS(filepath);
to keep it consistent with the original one.